### PR TITLE
Added jupyter-notebook version of the numpy example

### PR DESCRIPTION
**Pull request by AnupJoseph**
Added the numpy air quality analysis example to a jupyter notebook

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: examples/numpy/air-quality-analysis/analysis_notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 3 - Define your new Hamilton functions & curate them into a TemporaryFunctionModule object.\n",
+    "# This enables you to add functions to your DAG without creating a proper module.\n",
+    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module.\n",
+    "\n",
+    "\n",
+    "# We'll take out all the private functions into their own separate module\n",
```
Comment (skrawcz) [inline examples/numpy/air-quality-analysis/analysis_notebook.ipynb:56]: In Hamilton the convention is functions with an underscore are "private". So you don't want to stick them in a module :)

So these are fine as they are -- just remove adding them to the module (and thus the driver). Things should just work.

**Review by skrawcz — CHANGES_REQUESTED** [review]
Looking good. Just one change around the private functions. No need to put them into a module.

File: examples/numpy/air-quality-analysis/analysis_notebook.ipynb (hunk 1 above)
Comment (AnupJoseph) [reply examples/numpy/air-quality-analysis/analysis_notebook.ipynb]: Ah okay, I misunderstood the idea behind the temporary module technique, I assumed it was about logical separation. Have removed it and it works as before. 

I have also changed the name of the notebook to `hamilton_notebook.ipynb` so as to keep it consistent across all examples.

File: examples/numpy/air-quality-analysis/hamilton_notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,1120 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 4 - Define your new Hamilton functions & curate them into a TemporaryFunctionModule object.\n",
+    "# This enables you to add functions to your DAG without creating a proper module.\n",
+    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module.\n",
```
Comment (skrawcz) [inline examples/numpy/air-quality-analysis/hamilton_notebook.ipynb:137]:
```suggestion
    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module, or use `%%writefile`\n",
```

File: examples/numpy/air-quality-analysis/hamilton_notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,1120 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 5 - Define your new Hamilton functions & curate them into a TemporaryFunctionModule object.\n",
+    "# This enables you to add functions to your DAG without creating a proper module.\n",
+    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module.\n",
```
Comment (skrawcz) [inline examples/numpy/air-quality-analysis/hamilton_notebook.ipynb:265]:
```suggestion
    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module, or use `%%writefile`.\n",
```

**Review by skrawcz — APPROVED** [review]
Looks good! Thanks.

**Review by skrawcz — APPROVED** [review]

**Pull request merged**